### DISCUSSION TITLE
[BUG] Try too many times to connect an unconnected peer

### DISCUSSION THREAD

**Discussion by liangping** [Chia-Network/chia-blockchain#5334] [Support]
**Describe the bug**
As we know, the introducer will introduce new peers to fullnode. However, fullnode will try to connect to one peer(114.84.211.13:8444) all the time( 2+ days in my test), and won't try to connect any another new peers. Consequently,  connections of fullnode will reduce eventually.

It seems that only happens on SSL connections:
`Cannot connect to host 114.84.211.13:8444 ssl:<ssl.SSLContext object at 0x7f856efd4ec0> [Connect call failed ('114.84.211.13', 8444)]`

**To Reproduce**

!. `tail -f .chia/mainnet/log/debug.log`

**Expected behavior**
Stop trying to connect to failed host and try a new one if failed connections reach a threshold number, say 50.

- OS: [e.g. Ubuntu]

**Comment by Sabcoll** [comment]
+1 can confirm.

**Comment by SirVladislav** [comment]
the same
everything was fine yesterday
today after reboot GUI logs ->

2021-05-17T15:32:01.745 full_node full_node_server        : INFO     Cannot connect to host 183.220.75.145:8444 ssl:<ssl.SSLContext object at 0x0000018F31062908> [Connect call failed ('183.220.75.145', 8444)]
2021-05-17T15:33:11.340 full_node full_node_server        : INFO     Cannot connect to host 78.46.22.33:8444 ssl:<ssl.SSLContext object at 0x0000018F3C982F98> [Connect call failed ('78.46.22.33', 8444)]
2021-05-17T15:33:14.128 full_node full_node_server        : INFO     -> handshake to peer 183.131.157.240 9d6497905ac2c649c76b8adb7f13ccca7ac0403a587a221f2a91551cf1a76a8f
2021-05-17T15:33:14.375 full_node full_node_server        : INFO     Connected with full_node {'host': '183.131.157.240', 'port': 8444}
2021-05-17T15:33:14.378 full_node full_node_server        : INFO     -> new_peak to peer 183.131.157.240 9d6497905ac2c649c76b8adb7f13ccca7ac0403a587a221f2a91551cf1a76a8f
2021-05-17T15:33:14.381 full_node full_node_server        : INFO     Connection closed: 183.131.157.240, node id: 9d6497905ac2c649c76b8adb7f13ccca7ac0403a587a221f2a91551cf1a76a8f
2021-05-17T15:33:14.383 full_node chia.full_node.full_node: INFO     peer disconnected {'host': '183.131.157.240', 'port': 8444}

**Comment by tojefest** [comment]
+1 also

**Comment by casa10** [comment]
> 
> 
> the same
> everything was fine yesterday
> today after reboot GUI logs ->
> 
> 2021-05-17T15:32:01.745 full_node full_node_server : INFO Cannot connect to host 183.220.75.145:8444 ssl:<ssl.SSLContext object at 0x0000018F31062908> [Connect call failed ('183.220.75.145', 8444)]
> 2021-05-17T15:33:11.340 full_node full_node_server : INFO Cannot connect to host 78.46.22.33:8444 ssl:<ssl.SSLContext object at 0x0000018F3C982F98> [Connect call failed ('78.46.22.33', 8444)]
> 2021-05-17T15:33:14.128 full_node full_node_server : INFO -> handshake to peer 183.131.157.240 9d6497905ac2c649c76b8adb7f13ccca7ac0403a587a221f2a91551cf1a76a8f
> 2021-05-17T15:33:14.375 full_node full_node_server : INFO Connected with full_node {'host': '183.131.157.240', 'port': 8444}
> 2021-05-17T15:33:14.378 full_node full_node_server : INFO -> new_peak to peer 183.131.157.240 9d6497905ac2c649c76b8adb7f13ccca7ac0403a587a221f2a91551cf1a76a8f
> 2021-05-17T15:33:14.381 full_node full_node_server : INFO Connection closed: 183.131.157.240, node id: 9d6497905ac2c649c76b8adb7f13ccca7ac0403a587a221f2a91551cf1a76a8f
> 2021-05-17T15:33:14.383 full_node chia.full_node.full_node: INFO peer disconnected {'host': '183.131.157.240', 'port': 8444}

Same problems for more and more.
You are connected to a peer abnd suddenly this error and they disconnect one by one until you are unsynced.
You can wait hours, nothing happen, no reconnection automatically.
Need to restart but problem reappear. 
Put manually peer ip address but problem reappear.
Often same issue is signaled to the support.

**Comment by SirVladislav** [comment]
Got a solution
If you have more then 1 active devices you need to connect them correctly using wallet keys 
try this -> https://www.youtube.com/watch?v=_WD9ZF62kvs

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
